File: Solving-Foundations-of-Programming/src/main/Interpret.java
package main;

/*
 * https://codingbat.com/prob/p294185
 * Write a simple interpreter which understands "+", "-", and "*" operations. 
 * Apply the operations in order using command/arg pairs starting with the initial value of `value`. 
 * If you encounter an unknown command, return -1.
 * 
 * interpret(1, ["+"], [1]) → 2
 * interpret(4, ["-"], [2]) → 2
 * interpret(1, ["+", "*"], [1, 3]) → 6
 * 
 * */
public class Interpret {
	public int cal(int value, String[] commands, int[] args_int) {
		for (int i = 0; i < commands.length; i++) {
			switch (commands[i]) {
				case "+":
					value += args_int[i];
					break;
				case "-":
					value -= args_int[i];
					break;
				case "*":
					value *= args_int[i];
					break;
				default:
					return -1;
			}	
		}
		return value;
	}
}

File: problem-solving-with-python/interpret.py
# https://codingbat.com/prob/p234011
# Write a simple interpreter which understands "+", "-", and "*" operations.
# Apply the operations in order using command/arg pairs starting with the initial value of `value`.
# If you encounter an unknown command, return -1.
#
# interpret(1, ['+'], [1]) → 2
# interpret(4, ['-'], [2]) → 2
# interpret(1, ['+', '*'], [1, 3]) → 6
def interpret(value, commands, args):
    for i in range(len(commands)):
        if commands[i] == "+":
            value += args[i]
        elif commands[i] == "-":
            value -= args[i]
        elif commands[i] == "*":
            value *= args[i]
        else:
            return -1
    return value

value = 1
commands = ["+", "*"]
args = [1, 3]
result = interpret(value, commands, args)
print(result)